traits/trait_numeric.py

```diff
@@ -351,13 +365,29 @@ class CArray(AbstractArray):
         second dimension must be at least 2.)
     value : numpy array
         A default value for the array.
+    casting : str
+        Casting rule for ``numpy.ndarray.astype``. Values that cannot be
+        cast to a supplied ``dtype`` according to this rule cannot be
+        assigned. Default is "unsafe".
     """
 
     def __init__(
-        self, dtype=None, shape=None, value=None, typecode=None, **metadata
+        self,
+        dtype=None,
+        shape=None,
+        value=None,
+        typecode=None,
+        casting="unsafe",
```

Comment:
Consider making this parameter keyword-only (here and elsewhere)? I don't think there's any good case for passing this by position.